### DISCUSSION TITLE
swanctl.conf shared configuration like stroke available?

### DISCUSSION THREAD

**Discussion by psalty** [strongswan/strongswan#1588] [Q&A] [answered]
Hi.

in stroke with ipsec.conf there is a special name for the connection %default that can define the default value of all the connections.

Is there equivalent name for swanctl connection name as well?

**Comment by tewarivarun** [comment, marked as answer]
Yes, you can define a connection "conn-default" and "child-default"
Later you use this conn-default in your real tunnel definition.


# default configuration for all conns
conn-defaults {
        version=2
        encap=yes
}
# default configuration for all child
child-defaults {
        mode=tunnel
        start_action=none
        dpd_action=clear
}

Later when defining the tunnel

tunnel-1 : conn-defaults {
                children {
                        tunnel-1-child-1 : child-defaults {
                        dpd_action=start
                        }
                }
        }

**Reply by psalty** [reply]
Thank you
